tests/integration/src/install_tests.rs

```diff
@@ -0,0 +1,1009 @@
+            ),
+        }],
+        user_groups: vec![
+            InitUserGroupInput {
+                id: Some(ADMIN_GROUP_ID.hyphenated().to_string()),
+                name: "admin".to_string(),
+            },
+        ],
+        named_rules: vec![],
+        disaster_recovery_committee: Some(DisasterRecoveryCommitteeDTO {
+            quorum: 1,
+            user_group_id: ADMIN_GROUP_ID.hyphenated().to_string(),
+        }),
+    }
+})]
+fn install_with_bad_input(#[case] bad_input: station_api::InitialConfig) {
+    let TestEnv {
+        env, controller, ..
+    } = setup_new_env();
+
+    let canister_id = env.create_canister_with_settings(Some(controller), None);
+
+    env.set_controllers(canister_id, Some(controller), vec![canister_id, controller])
+        .expect("failed to set canister controller");
+
+    env.add_cycles(canister_id, 5_000_000_000_000);
+    let station_wasm = get_canister_wasm("station").to_vec();
+    let upgrader_wasm = get_canister_wasm("upgrader").to_vec();
+
+    let station_init_args = SystemInstall::Init(SystemInit {
+        name: "Station".to_string(),
+        upgrader: station_api::SystemUpgraderInput::Deploy(
+            station_api::DeploySystemUpgraderInput {
+                wasm_module: upgrader_wasm.clone(),
+                initial_cycles: Some(1_000_000_000_000),
+            },
+        ),
+        fallback_controller: Some(controller),
+        initial_config: bad_input,
+    });
+    env.install_canister(
+        canister_id,
+        station_wasm.clone(),
+        Encode!(&station_init_args).unwrap(),
+        Some(controller),
+    );
+
+    await_station_healthy(&env, canister_id, WALLET_ADMIN_USER);
+}
+
+fn assert_initial_users(
+    env: &PocketIc,
+    canister_id: Principal,
+    requester: Principal,
+    expected_users: &Vec<UserInitInput>,
+    default_groups: &Vec<UuidDTO>,
+) -> Result<(), String> {
+    let listed_users = list_users(env, canister_id, requester)
+        .expect("failed to get users")
+        .0
+        .expect("failed to get users");
+
+    if expected_users.len() != listed_users.users.len() {
+        return Err(format!(
+            "expected {} users, got {}",
+            expected_users.len(),
+            listed_users.users.len()
+        ));
+    }
+
+    for expected_user in expected_users {
+        let user = listed_users
+            .users
+            .iter()
+            .find(|user| user.name == expected_user.name)
+            .ok_or(format!("user {} not found", expected_user.name))?;
+
+        expected_user.identities.iter().all(|identity| {
+            user.identities
+                .iter()
+                .any(|user_identity| user_identity == &identity.identity)
+        });
+
+        expected_user
+            .groups
+            .as_ref()
+            .unwrap_or(default_groups)
+            .iter()
+            .all(|group| user.groups.iter().any(|user_group| &user_group.id == group));
```

Comment:
this is presumably ignored because it isn't in an assertion (same above)

Reply:
nice catch!